src/server/game/Globals/AreaTriggerDataStore.cpp

```diff
@@ -197,6 +199,47 @@ void AreaTriggerDataStore::LoadAreaTriggerTemplates()
         TC_LOG_INFO("server.loading", ">> Loaded 0 Spell AreaTrigger templates. DB table `spell_areatrigger` is empty.");
     }
 
+    //                                                                  0            1             2                3             4        5                 6
+    if (QueryResult circularMovementInfos = WorldDatabase.Query("SELECT SpellMiscId, CircleRadius, BlendFromRadius, InitialAngle, ZOffset, CounterClockWise, CanLoop FROM `spell_areatrigger_circular` ORDER BY `SpellMiscId`"))
+    {
+        do
+        {
+            Field* circularMovementInfoFields = circularMovementInfos->Fetch();
+            uint32 spellMiscId = circularMovementInfoFields[0].GetUInt32();
+
+            auto atSpellMiscItr = _areaTriggerTemplateSpellMisc.find(spellMiscId);
+
+            if (atSpellMiscItr == _areaTriggerTemplateSpellMisc.end())
+                continue;
+
+            AreaTriggerCircularMovementInfo& circularMovementInfo = atSpellMiscItr->second.CircularMovementInfo;
+
+#define VALIDATE_AND_SET_FLOAT(Float, Value) \
+            circularMovementInfo.Float = Value; \
+            if (!std::isfinite(circularMovementInfo.Float)) \
```

Comment:
is it possible that we get not finite values from db?

Reply:
It shouldn't be possible, but these are the checks from the client.. and as shauren said, if they're not "valid" by some reasons, there might be a client crash.. shauren wanted the check, even if its not possible.. but yeah :)